gpslogger/src/test/java/com/mendhak/gpslogger/loggers/MockLocations.java

```diff
@@ -40,43 +41,43 @@ public Location build(){
     }
 
     public MockLocations withAltitude(double altitude) {
-        when(loc.hasAltitude()).thenReturn(true);
-        when(loc.getAltitude()).thenReturn(altitude);
+        lenient().when(loc.hasAltitude()).thenReturn(true);
+        lenient().when(loc.getAltitude()).thenReturn(altitude);
         return this;
     }
 
     public MockLocations withBearing(float bearing) {
-        when(loc.hasBearing()).thenReturn(true);
-        when(loc.getBearing()).thenReturn(bearing);
+        lenient().when(loc.hasBearing()).thenReturn(true);
+        lenient().when(loc.getBearing()).thenReturn(bearing);
         return this;
     }
 
 
     public MockLocations withSpeed(float speed) {
-        when(loc.hasSpeed()).thenReturn(true);
-        when(loc.getSpeed()).thenReturn(speed);
+        lenient().when(loc.hasSpeed()).thenReturn(true);
+        lenient().when(loc.getSpeed()).thenReturn(speed);
         return this;
     }
 
     public MockLocations withTime(long date){
-        when(loc.getTime()).thenReturn(date);
+        lenient().when(loc.getTime()).thenReturn(date);
         return this;
     }
 
     public MockLocations putExtra(String k, String v) {
 
-        when(loc.getExtras()).thenReturn(bundle);
-        when(bundle.getString(eq(k))).thenReturn(v);
-        when(bundle.getString(eq(k),anyString())).thenReturn(v);
+        lenient().when(loc.getExtras()).thenReturn(bundle);
+        lenient().when(bundle.getString(eq(k))).thenReturn(v);
+        lenient().when(bundle.getString(eq(k),anyString())).thenReturn(v);
 
         return this;
     }
 
     public MockLocations putExtra(String k, int v) {
 
-        when(loc.getExtras()).thenReturn(bundle);
-        when(bundle.getInt(eq(k))).thenReturn(v);
-        when(bundle.getInt(eq(k),anyInt())).thenReturn(v);
+        lenient().when(loc.getExtras()).thenReturn(bundle);
+        lenient().when(bundle.getInt(eq(k))).thenReturn(v);
+        lenient().when(bundle.getInt(eq(k),anyInt())).thenReturn(v);
```

Comment:
Mockito's [strict stubbing](https://www.baeldung.com/mockito-unnecessary-stubbing-exception) doesn't detect reusable or generic use cases.  So I have to add a lenient() in front.